client.js

```diff
@@ -375,65 +333,55 @@ write-data=${ instance.path() }\r\n
 		"username": config.username,
 		"token": config.token,
 		"game_password": config.game_password,
-		"verify_user_identity": config.verify_user_identity,
-		"admins": [config.username],
+		"require_user_verification": config.verify_user_identity,
 		"allow_commands": config.allow_commands,
 		"autosave_interval": 10,
 		"autosave_slots": 5,
 		"afk_autokick_interval": 0,
 		"auto_pause": config.auto_pause,
 	};
-	fs.writeFileSync(instance.path("server-settings.json"), JSON.stringify(serversettings, null, 4));
-    console.log("Server settings: "+JSON.stringify(serversettings, null, 4));
-    console.log("Creating save .....");
-    let factorio = child_process.spawn(
-        './' + config.factorioDirectory + '/bin/x64/factorio', [
-            '-c', instance.path("config.ini"),
-            // '--create', instance.path("saves", "save.zip"),
-            '--start-server-load-scenario', 'Hotpatch',
-            '--server-settings', instance.path("server-settings.json"),
-            '--rcon-port', Number(process.env.RCONPORT) || instconf.clientPort,
-            '--rcon-password', instconf.clientPassword,
-        ], {
-			'stdio': ['pipe', 'pipe', 'pipe'],
-			'detached': process.platform === "linux",
-        }
-    );
-    factorio.stdout.on("data", data => {
-        data = data.toString("utf8").replace(/(\r\n\t|\n|\r\t)/gm,"");
-        console.log(data);
-        if(data.includes("Starting RCON interface")){
-            let client = new Rcon({
-				packetResponseTimeout: 200000,
-			    maxPending: 5
-			});
-            client.connect({
-                host: 'localhost',
-                port: Number(process.env.RCONPORT) || instconf.clientPort,
-                password: instconf.clientPassword
-            });
-            client.onDidAuthenticate(() => {
-                console.log('Clusterio | RCON Authenticated!');
-            });
-            client.onDidConnect(() => {
-                console.log('Clusterio | RCON Connected, starting save');
-                client.send("/c game.server_save('hotpachSave')");
-            });
-        }
-        if(data.includes("Saving finished")){
-            console.log("Map saved as hotpachSave.zip, exiting...");
-            console.log("Instance created!")
-            process.exit(0);
-        }
-		if(data.includes("Downloading from auth server failed")){
-			console.error("Instance creation failed, unable to establish auth server connection.");
-			console.error("Deleting broken instance...");
-			
-			fileOps.deleteFolderRecursiveSync(path.resolve(config.instanceDirectory, process.argv[3]));
-			
-			process.exit(0);
+
+	for (let [name, value] of Object.entries(overrides)) {
+		if (!Object.hasOwnProperty.call(serverSettings, name)) {
+			throw Error(`Expected server settings to have a ${name} property`);
 		}
+		serverSettings[name] = value;
+	}
+
+	await fs.writeFile(instance.path("server-settings.json"), JSON.stringify(serverSettings, null, 4));
```

Comment:
filename should be a `const server_config = "server-settings.json";`  or something like that. We have been working on removing hard-coded file names from the codebase.

Reply:
The thing is - you define the file names somewhere at the top of the file, preferably the root file and you just use that variable throughout the code. And it makes it easier to also add the ability to pass the file name as a command-line parameter. While we use instance-per-dir approach, you might want to use more than one server-settings.json file or even store them somewhere else and pass them at launch as paths.